yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtilSuite.scala

```diff
@@ -245,4 +247,28 @@ class YarnSparkHadoopUtilSuite extends SparkFunSuite with Matchers with Logging
       System.clearProperty("SPARK_YARN_MODE")
     }
   }
+
+  test("Obtain tokens For HiveMetastore") {
+    val hadoopConf = new Configuration()
+    hadoopConf.set("hive.metastore.kerberos.principal", "bob")
+    // thrift picks up on port 0 and bails out, without trying to talk to endpoint
+    hadoopConf.set("hive.metastore.uris", "http://localhost:0")
+    val util = new YarnSparkHadoopUtil
+    val e = intercept[InvocationTargetException] {
+      val token = util.obtainTokenForHiveMetastoreInner(hadoopConf, "alice")
+      fail(s"Expected an exception, got the token $token")
+    }
+    val inner = e.getCause
+    if (inner == null) {
+      fail("No inner cause", e)
+    }
+    if (!inner.isInstanceOf[HiveException]) {
+      fail(s"Not a hive exception", inner)
```

Comment:
I want to include the inner exception if it's of the wrong type, so that the Junit/jenkins report can diagnose the failure. an `assert(inner.isInstanceOf[HiveException])` will say when the exception was of the wrong type, but not include the details, including the stack trace.